File: mvvmLib/src/main/java/com/goldze/mvvmhabit/message/receiver/ScreenReceiver.java
package com.goldze.mvvmhabit.message.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.goldze.mvvmhabit.message.bus.RxBus;
import com.goldze.mvvmhabit.message.status._ScreenStatus;

/**
 * ClsFunction：是否息屏
 * CreateDate：2024/3/20
 * Author：TimeWillRememberUs
 */
public class ScreenReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (Intent.ACTION_SCREEN_OFF.equals(action)) {
            RxBus.getDefault().post(new _ScreenStatus(true));
        } else if (Intent.ACTION_SCREEN_ON.equals(action)) {
            RxBus.getDefault().post(new _ScreenStatus(false));
        }
    }
}

File: KotlinMvvmLib/src/main/java/com/maunc/jetpackmvvm/receiver/ScreenReceiver.kt
package com.maunc.jetpackmvvm.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.kunminx.architecture.ui.callback.UnPeekLiveData

/**
 *ClsFunction：
 *CreateDate：2024/5/26
 *Author：TimeWillRememberUs
 */
class ScreenReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (Intent.ACTION_SCREEN_OFF == action) {
            ScreenStateManager.instance.mScreenState.value = false
        } else if (Intent.ACTION_SCREEN_ON == action) {
            ScreenStateManager.instance.mScreenState.value = true
        }
    }
}

class ScreenStateManager private constructor() {

    val mScreenState = UnPeekLiveData<Boolean>()

    companion object {
        val instance: ScreenStateManager by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) {
            ScreenStateManager()
        }
    }
}